src/QuadForm/Herm/Lattices.jl

```diff
@@ -16,136 +16,103 @@ end
-    z.pmat = B
-    z.gram = gram
-    z.involution = involutionL
-    z.base_algebra = K
-    return z
+function hermitian_lattice(E::NumField, B::PMat ; gram = nothing)
+  @req rank(matrix(B)) == min(nrows(B), ncols(B)) "B must be of full rank"
+  @req nf(base_ring(B)) == E "Incompatible arguments: B must be defined over E"
+  @req degree(E) == 2 "E must be a quadratic extension"
+  if gram === nothing
+    V = hermitian_space(E, ncols(B))
+  else 
+    @assert gram isa MatElem
+    @req issquare(gram) "gram must be a square matrix"
+    @req ncols(B) == nrows(gram) "Incompatible arguments: the number of columns of B must correspond to the size of gram"
+    gram = map_entries(E, gram)
+    V = hermitian_space(E, gram)
   end
+  return lattice(V, B)
 end
 
 @doc Markdown.doc"""
-    hermitian_lattice(K::NumField, B::MatElem; gram_ambient_space = F) -> HermLat
+    hermitian_lattice(E::NumField, basis::MatElem ; gram = M) -> HermLat
 
-Given a number field $K$ and a matrix $B$, returns the hermitian lattice
-spanned by the rows of $B$ inside the hermitian space over $K$ with gram matrix
-$F$.
+Given a matrix `basis` and a number field `E` of degree 2, return the hermitian lattice
+spanned by the rows of `basis` inside the hermitian space over $E$ with gram matrix $M$.
 
-If $F$ is not supplied, the gram matrix of the ambient space will be the
-identity matrix.
+If $M$ is not supplied, the gram matrix of the ambient space will be the identity
+matrix over $E$ of size the number of columns of `basis`.
 """
-hermitian_lattice(::NumField, ::MatElem; gram_ambient_space = nothing)
+hermitian_lattice(E::NumField, basis::MatElem ; gram = nothing) = hermitian_lattice(E, pseudo_matrix(basis), gram = gram)
 
-function hermitian_lattice(K::NumField, B::MatElem; gram_ambient_space = nothing, gram = nothing)
-  if gram_ambient_space === nothing && gram === nothing
-    return HermLat(K, identity_matrix(K, nrows(B)), pseudo_matrix(B))
-  end
-  if gram_ambient_space !== nothing && gram === nothing
-    return HermLat(K, gram_ambient_space, pseudo_matrix(B))
-  end
-  if gram_ambient_space === nothing && gram !== nothing
-    involutionL = involution(K)
-
-    P = pseudo_matrix(B)
-    z = HermLat{typeof(K), typeof(base_field(K)), typeof(B), typeof(P), morphism_type(typeof(K))}()
-    z.pmat = P
-    z.gram = gram
-    z.involution = involutionL
-    z.base_algebra = K
-    return z
+@doc Markdown.doc"""
+    hermitian_lattice(E::NumField, gens::Vector ; gram = M) -> HermLat
+
+Given a list of vectors `gens` and a number field `E` of degree 2, return the hermitian
+lattice spanned by the elements of `gens` inside the hermitian space over $E$ with 
+gram matrix $M$.
+
+If $M$ is not supplied, the gram matrix of the ambient space will be the identity
+matrix over $E$ of size the length of the elements of `gens`.
+"""
+function hermitian_lattice(E::NumField, gens::Vector ; gram = nothing) 
+  if length(gens) == 0
```

Comment:
This part does not seem to make sense. If the generators are of length zero, then the gram matrix must be a 0x0 matrix.

Reply:
This is the way used to create the zero lattice. And `gram` in these constructors always refers to the gram matrix of the ambient space, so one could create the zero lattice in a non trivial space, no ?

Reply:
Of course, you are right.